plugin.video.viaplay/resources/lib/viaplay.py

```diff
@@ -0,0 +1,347 @@
+﻿# -*- coding: utf-8 -*-
+"""
+A Kodi-agnostic library for Viaplay
+"""
+import codecs
+import os
+import cookielib
+import calendar
+import re
+import json
+import uuid
+import HTMLParser
+from collections import OrderedDict
+from datetime import datetime, timedelta
+
+import iso8601
+import requests
+
+
+class Viaplay(object):
+    def __init__(self, settings_folder, country, debug=False):
+        self.debug = debug
+        self.country = country
+        self.settings_folder = settings_folder
+        self.cookie_jar = cookielib.LWPCookieJar(os.path.join(self.settings_folder, 'cookie_file'))
+        self.tempdir = os.path.join(settings_folder, 'tmp')
+        if not os.path.exists(self.tempdir):
+            os.makedirs(self.tempdir)
+        self.deviceid_file = os.path.join(settings_folder, 'deviceId')
+        self.http_session = requests.Session()
+        self.device_key = 'xdk-%s' % self.country
+        self.base_url = 'https://content.viaplay.{0}/{1}'.format(self.country, self.device_key)
+        self.login_api = 'https://login.viaplay.%s/api' % self.country
+        try:
+            self.cookie_jar.load(ignore_discard=True, ignore_expires=True)
+        except IOError:
+            pass
+        self.http_session.cookies = self.cookie_jar
+
+    class ViaplayError(Exception):
+        def __init__(self, value):
+            self.value = value
+
+        def __str__(self):
+            return repr(self.value)
+
+    def log(self, string):
+        if self.debug:
+            try:
+                print '[Viaplay]: %s' % string
```

Comment:
`print('[Viaplay]: %s' % string)` (if you can't use `xbmc.log(...)` here) for python 3 compatibility